src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Remove.cs

```diff
@@ -0,0 +1,579 @@
+                            {
+                                fValid = false;
+                                errors.EnsureError(DocumentErrorSeverity.Severe, args[i], TexlStrings.ErrRemoveAllArg);
+                            }
+                        }
+
+                        continue;
+                    }
+                    else
+                    {
+                        fValid = false;
+                        errors.EnsureError(args[i], TexlStrings.ErrNeedRecord_Arg, args[i]);
+                        continue;
+                    }
+                }
+
+                var collectionAcceptsRecord = collectionType.Accepts(argType.ToTable(), exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: context.Features.PowerFxV1CompatibilityRules);
+                var recordAcceptsCollection = argType.ToTable().Accepts(collectionType, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: context.Features.PowerFxV1CompatibilityRules);
+
+                // The item schema should be compatible with the collection schema.
+                if (!collectionAcceptsRecord && !recordAcceptsCollection)
+                {
+                    fValid = false;
+                    if (!SetErrorForMismatchedColumns(collectionType, argType, args[i], errors, context.Features))
+                    { 
+                        errors.EnsureError(DocumentErrorSeverity.Severe, args[i], TexlStrings.ErrCollectionDoesNotAcceptThisType);
+                    }
+                }
+                
+                // Only warn about no-op record inputs if there are no data sources that would use reference identity for comparison.
+                else if (!collectionType.AssociatedDataSources.Any() && !recordAcceptsCollection)
+                {
+                    errors.EnsureError(DocumentErrorSeverity.Warning, args[i], TexlStrings.ErrCollectionDoesNotAcceptThisType);
+                }
+
+                if (!context.AnalysisMode)
+                {
+                    // ArgType[N] (0<N<argCount) must match all the fields with the data source.
+                    bool checkAggregateNames = argType.CheckAggregateNames(collectionType, args[i], errors, context.Features, SupportsParamCoercion);
+
+                    // The item schema should be compatible with the collection schema.
+                    if (!checkAggregateNames)
+                    {
+                        fValid = false;
+                        if (!SetErrorForMismatchedColumns(collectionType, argType, args[i], errors, context.Features))
+                        {
+                            errors.EnsureError(DocumentErrorSeverity.Severe, args[i], TexlStrings.ErrTableDoesNotAcceptThisType);
+                        }
+                    }
+                }
+            }
+
+            if (context.AnalysisMode)
```

Comment:
>AnalysisMode [](http://example.com/codeflow?start=24&length=12)

This definitely should not be context.AnalysisMode, but `!context.Features.PowerFxV1CompatibilityRules` instead

Reply:
The original code wasn't checking if `context.Features.PowerFxV1CompatibilityRules` was active or not. The idea here is, whatever PA was returning, keep returning it. Since PFxV1 in Core is active, return `Void`.

Reply:
When Power Apps moves to PFxV1, it should have the same behavior as the C# interpreter. Remove is one of the functions that we will update the return type on V1 to be Void.